python/pyspark/sql/functions.py

```diff
@@ -208,6 +208,14 @@ def sumDistinct(col):
     return _invoke_function_over_column("sumDistinct", col)
 
 
+@since(3.2)
+def product(col, scale=1.0):
+    """
+    Aggregate function: returns the product of the values in a group.
+    """
```

Comment:
Let's also add some doctests here.

Reply:
I agree that having some documentation, with examples, would be very helpful and I've added these following your advice. However, it seems like the automated build tests are insanely restrictive in their validation of these doctests, causing multiple building failures when the pydoc examples differ by inconsequential whitespace or formatting changes. For example, even when I have directly copy-pasted example outputs from a pyspark session, these were rejected by the build-system.

Can I suggest that someone investigates whether the build-system can make these documentation checks more tolerant of whitespace variations, which I'd expect will trip-up many developers?